Cargo.toml

```diff
@@ -29,6 +29,7 @@ once_cell = "1"
 sg-gql = { path  = "crates/sg-gql" }
 sg-types = { path  = "crates/sg-types" }
 jsonrpc = { path = "crates/jsonrpc" }
+gix = "0.52.0"
```

Comment:
This adds a dependency on `gix` for all the git stuff. Feels worth it, but it is a somewhat large dependency.